cmd/gardenlet/app/app.go

```diff
@@ -464,15 +477,66 @@ func (g *garden) registerSeed(ctx context.Context, gardenClient client.Client) e
 	})
 }
 
+func (g *garden) migrateAllShootServicesForNetworkPolicies(ctx context.Context, log logr.Logger) error {
+	var (
+		taskFns                  []flow.TaskFn
+		kubeApiServerServiceList = &corev1.ServiceList{}
+	)
+
+	// Kube-Apiserver services
+	if err := g.mgr.GetClient().List(ctx, kubeApiServerServiceList, client.MatchingLabels{v1beta1constants.LabelApp: v1beta1constants.LabelKubernetes, v1beta1constants.LabelRole: v1beta1constants.LabelAPIServer}); err != nil {
+		return err
+	}
+
+	taskFns = append(taskFns, migrationTasksForServices(g.mgr.GetClient(), kubeApiServerServiceList.Items, kubeapiserverconstants.Port)...)
+
+	// VPN-Seed-Server services
+	for _, serviceName := range []string{vpnseedserver.ServiceName, fmt.Sprintf("%s-%d", vpnseedserver.ServiceName, 0), fmt.Sprintf("%s-%d", vpnseedserver.ServiceName, 0)} {
+		serviceList := &corev1.ServiceList{}
+		// Use APIReader here because an index on `metadata.name` is not available in the runtime client.
+		if err := g.mgr.GetAPIReader().List(ctx, serviceList, client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector(metav1.ObjectNameField, serviceName),
+		}); err != nil {
+			return err
+		}
+
+		taskFns = append(taskFns, migrationTasksForServices(g.mgr.GetClient(), serviceList.Items, vpnseedserver.MetricsPort)...)
```

Comment:
Shouldn't it be like this?

```suggestion
		taskFns = append(taskFns, migrationTasksForServices(g.mgr.GetClient(), serviceList.Items, vpnseedserver.OpenVPNPort)...)
```

Reply:
The port is used for the `all-scrape-targets` annotation. Why should we need the OpenVPN Port here?

Reply:
Right. My bad. Ignore this